runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -1286,6 +1286,16 @@ TR::SymbolValidationManager::validateStaticClassFromCPRecord(uint16_t classID, u
    {
    J9Class *beholder = getJ9ClassFromID(beholderID);
    J9ConstantPool *beholderCP = J9_CP_FROM_CLASS(beholder);
+
+   // This CP entry could still be unresolved, which would cause
+   // getClassOfStaticFromCP() to fail. Attempt to resolve it manually to reduce
+   // the chances of failure.
+   if (cpIndex != -1)
+      {
+      TR::VMAccessCriticalSection getClassFromConstantPool(_fej9);
+      _fej9->_vmFunctionTable->resolveStaticFieldRef(_fej9->vmThread(), NULL, beholderCP, cpIndex, J9_RESOLVE_FLAG_JIT_COMPILE_TIME, NULL);
```

Comment:
From the code, it looks like we call `jitCTResolveStaticFieldRefWithMethod` to do compile time resolution of static fields. Ideally we'd call the same thing, but we do have the problem of needing to poof up a `J9Method`, so `resolveStaticFieldRef` seems ok.

Reply:
This made me worry about whether the `J9Method` could affect the result, but I think it's fine.

The method is only used to determine whether it's OK to store to a static final field. If at compile-time we generated a resolved static field store, then the store was allowed at the time, so one of the following will hold:
- It was non-final. On (successful) load, it will still be non-final because we validate the class chain of the defining class.
- It was final and defined by the same class as the method containing the `putstatic`. On (successful) load, this will still be the case because the class ref in the static field ref CP entry names the containing class (i.e. the class that owns the CP entry), and we validate the class chain of the defining class of each inlined method. In this case the class file version must be <53 or the `putstatic` must occur within `<clinit>`, both of which are also captured in the class chain.

Also, when generating IL for `putstatic`, all of the above is independent of how we validate `classOfStatic()`, which is only used for generating a write barrier